llvm/test/Object/macho-rebase-bind-trailing-opcode-infinite-loop.test

```diff
@@ -0,0 +1,23 @@
+// A valid MachO object with a weak-bind table following a bind table ending
```

Comment:
nits: I think this comment should go into the input file instead of test case file because it seems to describe the properties of the input file vs how the test is constructed. 

Also, can you rename the test to something more like `macho-rebase-bind-trailing-opcode-boundary.test`?
The infinite loop part doesn't seem relevant besides that's how the bug manifested before which would be one but not all cases of how this test could `break` in the future.